connexion/decorators/parameter.py

```diff
@@ -108,34 +111,49 @@ def sanitize_param(name):
     def make_request_query(request):
         request_query = {}
         try:
-            for k, v in request.query.to_dict(flat=False).items():
+            for k, values in request.query.to_dict(flat=False).items():
                 k = sanitize_param(k)
-                query_param = query_types.get(k, None)
-                if query_param is not None and query_param["type"] == "array":
-                    if query_param.get("collectionFormat", None) == "pipes":
-                        request_query[k] = "|".join(v)
-                    else:
-                        request_query[k] = ",".join(v)
+                query_defn = query_defns.get(k, None)
+                query_schema = get_schema(query_defn)
+                if (query_schema is not None and query_schema['type'] == 'array'):
+                    request_query[k] = resolve_query_duplicates(values, query_defn)
                 else:
-                    request_query[k] = v[0]
+                    request_query[k] = values[-1]
         except AttributeError:
             request_query = {sanitize_param(k): v for k, v in request.query.items()}
         return request_query
 
+    # swagger2 body
     body_parameters = [parameter for parameter in parameters if parameter['in'] == 'body'] or [{}]
     body_name = sanitize_param(body_parameters[0].get('name'))
     default_body = body_parameters[0].get('schema', {}).get('default')
-    query_types = {sanitize_param(parameter['name']): parameter
+
+    form_defns = {sanitize_param(parameter['name']): parameter
+                  for parameter in parameters
+                  if parameter['in'] == 'formData'}
+
+    # openapi3 body
+    if body_name is None and body_schema is not None:
```

Comment:
what happens if body_name is blank?

Reply:
I don't _think_ body_name can be blank in swagger2 as "name" is a required field(https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameterObject). It is set to `None` in oas3 because `requestBody` replace the `in["body"]` parameters of swagger2, so `body_parameters = [{}]`.

I find all this interspersed logic really yucky, so I'm open to ideas for how to split it out by spec version.
This file is especially bad :disappointed: 
related: see my other branch that does this with the Operation class https://github.com/dtkav/connexion/compare/18a7c28...dtkav:oas3_class_separation#diff-f4018bfc9e3bdfe01ab150a7069d72ceL183